src/store/enterprise/managedStorage.ts

```diff
@@ -124,19 +143,47 @@ export async function watchDelayedStorageInitialization(): Promise<void> {
 // uBlock (still) contains a workaround to automatically reload the extension on initial install
 // - https://github.com/gorhill/uBlock/commit/32bd47f05368557044dd3441dcaa414b7b009b39
 const waitForInitialManagedStorage = pMemoize(async () => {
-  // Returns undefined if the promise times out
-  managedStorageSnapshot = await pollUntilTruthy<
-    Nullishable<ManagedStorageState>
-  >(readPopulatedManagedStorage, {
-    maxWaitMillis: MAX_MANAGED_STORAGE_WAIT_MILLIS,
-  });
+  if (await initializationTimestamp.get()) {
+    // The extension has waited already this session, so don't wait again
+    console.debug("Managed storage already initialized this session");
+    managedStorageSnapshot = (await readManagedStorageImmediately()) ?? {};
+  } else {
+    console.debug(
+      `Managed storage not initialized yet, polling for ${MAX_MANAGED_STORAGE_WAIT_MILLIS}ms`,
+    );
+    const waitController = new AbortController();
+    // Skip polling if it becomes initialized while waiting
+    initializationTimestamp.onChanged(
+      waitController.abort,
```

Comment:
```suggestion
      waitController.abort.bind(waitController),
```

But in reality this cannot happen. Line 185 (`initializationTimestamp.set`) cannot be reached by another process while this one is running because `waitForInitialManagedStorage` is only run once.

Reply:
I think that's related to the misunderstanding here? -- that the timestamp is using local storage to be available across contexts: https://github.com/pixiebrix/pixiebrix-extension/pull/7790#discussion_r1510253953

The abort logic in the `waitForInitialManagedStorage` handles the case where:
1. Background page starts up and is starts polling for managed storage
2. Extension Console Page Starts up and starts polling
3. Background page page finished polling
4. The new abort controller allows the Extension Console page quit polling because it knows the Background Page already polled the 4.5 seconds